tcms/testruns/tests/test_views.py

```diff
@@ -608,3 +608,83 @@ def test_show_add_cases_to_run(self):
             ]
             for html in html_pieces:
                 self.assertContains(response, html, html=True)
+
+
+class TestCaserunEditMenus(BaseCaseRun):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+    def setUp(self):
+        self.menu_perms = [
+            'testruns.change_testcaserun',
+            'testruns.add_testcaserun',
+            'testruns.delete_testcaserun',
+            'testcases.add_bug',
+            'testcases.delete_bug',
+            'django_comments.add_comment'
+        ]
+
+        self.menu_html = [
+            # Cases Menu
+
+            '<span>Cases</span>',
+            '<a href="{0}" class="addBlue9">Add</a>'
+            .format(reverse('add-cases-to-run', args=[self.test_run.pk])),
+
+            '<a href="#" title="Remove selected cases form this test run" \
+            data-param="{0}" \
+            class="removeBlue9 js-del-case">Remove</a>'.format(self.test_run.pk),
+
+            '<a href="#" title="Update the IDLE case runs to newest case text" \
+            href="javascript:void(0)" data-param="{0}" \
+            class="updateBlue9 js-update-case">Update</a>'
+            .format(reverse('testruns-update_case_run_text', args=[self.test_run.pk])),
+
+            '<a href="#" title="Assignee this case(s) to other people" \
+            class="assigneeBlue9 js-change-assignee">Assignee</a>',
+
+            # Status Menu
+
+            '<span>Status</span>',
+
+            # Bugs Menu
+
+            '<span>Bugs</span>',
+            '<a href="javascript:void(0);" class="addBlue9 js-add-bugs">Add</a>',
+            '<a href="javascript:void(0);" class="removeBlue9 js-remove-bugs">Remove</a>',
+
+            # Comment Menu
+
+            '<span>Comment</span>',
+            '<a href="#" class="addBlue9 js-show-commentdialog">Add</a>'
+        ]
+
+        self.url = reverse('testruns-get', args=[self.test_run.pk])
+        self.response = None
+
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.tester.username,
+            password='password')
+
+    def getNewResponse(self):
+        self.response = self.client.get(self.url)
+        self.assertEqual(HTTPStatus.OK, self.response.status_code)
+
+    def test_get_menus_with_permissions(self):
+        for perm_string in self.menu_perms:
+            user_should_have_perm(self.tester, perm_string)
+
+        self.getNewResponse()
+
+        for html_string in self.menu_html:
+            self.assertContains(self.response, html_string, html=True)
```

Comment:
Not very nice. These should be separate test methods so that when something fails we know what failed. Otherwise when this method fails we have to spend the time to debug the entire menu and its permissions to figure out which exactly part failed.

While at it the menu_html list should become separate variables for each part of the menu.